src/app/portlets/dot-apps/dot-apps-import-export-dialog/dot-apps-import-export-dialog.component.ts

```diff
@@ -0,0 +1,194 @@
+import {
+    Component,
+    OnDestroy,
+    Input,
+    OnChanges,
+    SimpleChanges,
+    ViewChild,
+    ElementRef,
+    Output,
+    EventEmitter
+} from '@angular/core';
+import { FormGroup, FormBuilder, FormControl, Validators } from '@angular/forms';
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { DotAppsService } from '@services/dot-apps/dot-apps.service';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import {
+    DotApps,
+    DotAppsExportConfiguration,
+    DotAppsImportConfiguration,
+    DotAppsSites
+} from '@shared/models/dot-apps/dot-apps.model';
+import { Subject } from 'rxjs';
+import { take, takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'dot-apps-import-export-dialog',
+    templateUrl: './dot-apps-import-export-dialog.component.html'
+})
+export class DotAppsImportExportDialogComponent implements OnChanges, OnDestroy {
+    @ViewChild('importFile') importFile: ElementRef;
+    @Input() action?: string;
+    @Input() app?: DotApps;
+    @Input() site?: DotAppsSites;
+    @Output() resolved: EventEmitter<boolean> = new EventEmitter();
+
+    showExportDialog = false;
+    form: FormGroup;
+    dialogActions: DotDialogActions;
+    errorMessage: string;
+    dialogHeaderKey = '';
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private dotAppsService: DotAppsService,
+        private dotMessageService: DotMessageService,
+        private fb: FormBuilder
+    ) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.action?.currentValue) {
```

Comment:
Di actions change after the component is created?

Reply:
yes, the user could EXPORT and then IMPORT, so the same dialog component is used and the action determines what template should show